processor/elasticinframetricsprocessor/go.mod

```diff
@@ -1,6 +1,6 @@
 module github.com/elastic/opentelemetry-collector-components/processor/elasticinframetricsprocessor
 
-go 1.21.1
+go 1.22.0
```

Comment:
@ishleenk17 Can you review this bit? The inframetrics processor changes require approval from an appropriate team member.